test/testdata/binaries/BUILD

```diff
@@ -38,5 +38,4 @@ apple_static_library(
 
 objc_library(
     name = "dummy_lib",
-    srcs = ["@bazel_tools//tools/objc:dummy.c"],
```

Comment:
Mmh, this is causing a failure due to `apple_static_library`:
```
ERROR: /Users/buildkite/builds/bk-imacpro-5/bazel/rules-apple-darwin/test/testdata/binaries/BUILD:32:21: Linking test/testdata/binaries/empty_staticlib-x86_64-apple-ios9.0-simulator-fl.a failed: (Exit 1): libtool failed: error executing command (from target //test/testdata/binaries:empty_staticlib)
--
  | (cd /private/var/tmp/_bazel_buildkite/df247ab1d8e1a2f86483eaa8de05c1e1/sandbox/darwin-sandbox/15/execroot/build_bazel_rules_apple && \
  | exec env - \
  | APPLE_SDK_PLATFORM=iPhoneSimulator \
  | APPLE_SDK_VERSION_OVERRIDE=15.0 \
  | PATH=/Users/buildkite/Library/Caches/bazelisk/downloads/bazelbuild/bazel-cc71db2a465716ba175d3defaa6032107bf1dd90-darwin-x86_64/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin \
  | XCODE_VERSION_OVERRIDE=13.0.0.13A233 \
  | ZERO_AR_DATE=1 \
  | external/local_config_cc/libtool -D -no_warning_for_no_symbols -static -arch_only x86_64 -syslibroot __BAZEL_XCODE_SDKROOT__ -o bazel-out/darwin-fastbuild/bin/test/testdata/binaries/empty_staticlib-x86_64-apple-ios9.0-simulator-fl.a)
  | # Configuration: 2025871da510c7dc8f9b9a3fabd34ddca5ef2b8cbc41dd18963857ec1999dac5
  | # Execution platform: @local_config_platform//:host
  |  
  | Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
  | error: /Applications/Xcode13.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: no files specified
```
My understanding is `apple_static_library` is slightly different from what we have at the moment and with our current implementation it fails when no inputs are specified, but we're blocked on https://github.com/bazelbuild/rules_apple/pull/1533. Is that correct? Is the preference here for wait to cherry-pick this change, or partially revert this part of the cherry-pick and go ahead with it? @keith @brentleyjones

Reply:
I'm not sure if that will fix this or not, but yea if reverting just this one works that's fine, since this patch is pretty low impact it's not a huge deal either way

Reply:
This partial revert has fixed it, so I'd say let's go ahead with it.